core/metacling/src/TClingClassInfo.h

```diff
@@ -89,7 +89,10 @@ class TClingClassInfo {
    void                 DeleteArray(void *arena, bool dtorOnly, const ROOT::TMetaUtils::TNormalizedCtxt &normCtxt) const;
    void                 Destruct(void *arena, const ROOT::TMetaUtils::TNormalizedCtxt &normCtxt) const;
    const clang::ValueDecl *GetDataMember(const char *name) const;
-   const clang::Decl      *GetDecl() const { return fDecl; } // Underlying representation without Double32_t
+   void SetDecl(const clang::Decl* D) {
+     // FIXME: We should track down all set and potentially avoid them.
+     fDecl = D;
```

Comment:
Shouldn't this invalidate the name caches?

Reply:
Yes, good point.